src/OpenTelemetry.Instrumentation.ServiceFabricRemoting/ActorRemoting/TraceContextEnrichedActorServiceV2RemotingDispatcher.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Text;
+using Microsoft.ServiceFabric.Actors.Remoting.V2.Runtime;
+using Microsoft.ServiceFabric.Actors.Runtime;
+using Microsoft.ServiceFabric.Services.Remoting.V2;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Runtime;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Instrumentation.ServiceFabricRemoting;
+
+/// <summary>
+/// Provides an implementation of Microsoft.ServiceFabric.Services.Remoting.V2.Runtime.IServiceRemotingMessageHandler
+/// that can dispatch messages to an actor service and to the actors hosted in the service
+/// </summary>
+public class TraceContextEnrichedActorServiceV2RemotingDispatcher : ActorServiceRemotingDispatcher
```

Comment:
Is it possible to implement TraceConstextEnriched\*RemotingDispatcher as a decorator of the IServiceRemotingMessageHandler interface instead of inheriting them two different SF classes?

Reply:
For the dispatchers I cannot use a decorator because the base classes are different, and the methods need to be overridden.

Reply:
ActorServiceRemotingDispatcher is derived from ServiceRemotingMessageDispatcher, which implements IServiceRemotingMessageHandler. Why can't a single TraceContextEnrichedRemotingDispatcher implement IServiceRemotingMessageHandler and take a constructor parameter of IServiceRemotingMessageHandler type, which could be either ActorServiceRemotingDispatcher or ServiceRemotingMessageDispatcher?

Reply:
@olegsych , I have replaced the two custom dispatchers with a single Dispatcher decorator. Please take a look at the latest code. Thanks.